Dockerfile.fedora42

```
@@ -0,0 +1,32 @@
+# Fedora 42 ships cmake 3.31, gcc 15, clang 20, glibc 2.41.
+#
+# Usage:
+#   docker-compose -f docker-compose.dev.yml build fedora-dev
+#   docker-compose -f docker-compose.dev.yml run --rm fedora-dev bash
+#
+# Inside the container:
+#   cmake -B build -DBUILD_TESTS=ON && cmake --build build && cd build && ctest
+
+FROM fedora:42
+
+RUN dnf install -y --setopt=install_weak_deps=False \
```

Hadolint (2.14.0)

[warning] Specify version with `dnf install -y <package>-<version>`.

(DL3041)